resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolverContext.java

```diff
@@ -510,11 +512,26 @@ private void followCname(InetSocketAddress nameServerAddr, String name, String c
 
         final InetSocketAddress nextAddr = nameServerAddrs.next();
         if (parent.isCnameFollowARecords()) {
-            query(nextAddr, new DefaultDnsQuestion(cname, DnsRecordType.A), promise);
+            if (!query(hostname, DnsRecordType.A, nextAddr, promise)) {
```

Comment:
I actually think its easier to read the way it is... So I would love to just leave it as it is. @Scottmitch thoughts ?